src/chargepoint/reservation/ReservationManager.cpp

```diff
@@ -227,13 +228,20 @@ bool ReservationManager::handleMessage(const ocpp::messages::ReserveNowReq& requ
                     connector->reservation_expiry_date   = request.expiryDate;
                     response.status                      = ReservationStatus::Accepted;
 
+                    if (connector->reservation_expiry_date > now)
+                    {
                     // Update connector status and notify new status
                     m_worker_pool.run<void>(
                         [this, connector]
                         {
                             m_status_manager.updateConnectorStatus(connector->id, ChargePointStatus::Reserved);
                             m_events_handler.reservationStarted(connector->id);
                         });
+                    }
+                    else
+                    {
+                        response.status = ReservationStatus::Rejected;
```

Comment:
The ocpp1.6 standard does not say anything about Rejecting a ReserveNow with an expiry date that lies in the past. I guess it would make sense when you think about it.

But in terms of sticking to the standard I would actually expect an accepted state, which is then quickly followed by a StatusNotficiation Available. De standard:
> When a reservation expires, the Charge Point SHALL terminate the reservation and make the connector
available. The Charge Point SHALL send a status notification to notify the Central System that the reserved
connector is now available.

What do you think?

Reply:
Yes, it is true, this issue is not mentioned in the standard, but in my opinion, this should never happen, and if it happens, it is due to a server mistake, and if we follow your suggested method, the server will never realize its mistake, and this mistake will seemas a normal routine.

Reply:
Of course, your opinion adheres more to the standard

Reply:
I would be in favor of sticking to the standard as much as possible.